torch/_dynamo/compiled_autograd.py

```diff
@@ -269,35 +286,59 @@ def set_node_origin(self, node_name, node_index):
         set_stack_trace(new_stack_trace)
 
 
-compiled_autograd_enabled = False
 override_compiler_fn: Optional[Callable[[Any], Any]] = None
 
 
+# Do not call this function directly, use `torch.compile` with a backend that invokes AOTAutograd, e.g. aot_eager, inductor
+def _initialize():
+    assert torch._dynamo.config.compiled_autograd_enabled
+    global override_compiler_fn
+    assert (
+        not override_compiler_fn
+    ), "Cannot torch.compile with compiled autograd within a compiled autograd context manager"
+
+    torch._C._dynamo.compiled_autograd.set_autograd_compiler(
+        AutogradCompilerInstance,
+        snapshot_verbose_logging_enabled(),
+        False,
+    )
+    # TODO: multithreading can only be re-enabled when compiled autograd is turned off and all nodes are freed
+    torch.autograd.set_multithreading_enabled(False)
+    compiled_autograd_log.info("Compiled autograd initialized.")
+
+
+# Forces .backward() call within to dispatch to compiled autograd
+# Allows for more fine-grained control over which autograd engine calls should be handled
 @contextlib.contextmanager
 def enable(compiler_fn):
-    global compiled_autograd_enabled, override_compiler_fn
-    assert not compiled_autograd_enabled
-    compiled_autograd_enabled = True
+    global override_compiler_fn
```

Comment:
Why is this called "override" are we overriding some other setting?  Or is this just the main way to set it?

Reply:
suppose we have 
```python
torch._dynamo.config.compiled_autograd = True
compiled_fn = torch.compile(fn, backend="inductor")  # _initialize()
loss = compiled_fn()
with enable(torch.compile(backend="aot_eager")):
  loss.backward()
```

backend aot_eager overrides the inductor compile wrapper set on the autograd nodes